### DISCUSSION TITLE
Use connectOrCreate for nested fields and relationships.

### DISCUSSION THREAD

**Discussion by techboy-coder** [prisma/prisma#7774] [Q&A] [answered]
Hey everybody. While working on my side project I got stuck.

This is my Prisma schema: 

```prisma
model Word {
  id        String   @id @default(uuid())
  text      String   @unique
  cards     Card[]
  createdAt DateTime @default(now())

  @@index([text])
}

model Card {
  id              String           @id @default(uuid())
  sides           Word[] //@unique <-- doesn't work
  card_collection CardCollection[]
  createdAt       DateTime         @default(now())

  @@index([id])
}

model CardCollection {
  id        String   @id @default(uuid())
  cards     Card[]
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@index([id])
}
```
I am using connectOrCreate to avoid duplicate relationships. 
When using connectOrCreate you first find if a relationship already exists (using where), and if it doesn't you create it.
The where clause inside connectOrCreate can only compare towards a unique field (such as a id). I want to compare to a field which is nested a level deeper.

Currently while creating a "card collection", I am looping through my input data, finding if a card with the given sides already exists, then getting its id, which I am finally using in the where clause (while creating using connect.

```typescript
const input_card_collection = [
    [
      { text: "cat" },
      { text: "Katze" },
      { text: "chat" },
    ],
    [
      { text: "dog" },
      { text: "Hund" },
      { text: "chien" },
    ],
    [
      { text: "to be" },
      { text: "sein" },
      { text: "être" },
    ]
  ]


  input_card_collection.map(async (single_card) => {
    const card = await prisma.card.findFirst({
      where: {
        AND: single_card.map((single_side) => {
          return { sides: { some: single_side } }
        })
      },
      include: {
        sides: true
      }
    })

    const cards = await prisma.cardCollection.create({
      data: {
        cards: {
          create: {
            card: {
              connectOrCreate: {
                create: {
                  sides: {
                    connectOrCreate: single_card.map((single_side) => {
                      return { create: single_side, where: single_side }
                    }),
                  },
                },
                where: {
                  id: card ? card.id : "null",
                  // Doesn't Work!
                  // sides: single_card
                }
              }
            }
          }
        }
      }
    })
  })
```
Though this code works, it seems very inefficient to me.  Is there a cleaner and more efficient way of doing this? It would be great if there were a way to refer to nested fields (not only 1 level but multiple levels deep), which finally leads to a unique field (In this example "text" from the "Word" model") Any thoughts or comments?

**Comment by ryands17** [comment, marked as answer]
@techboy-coder 👋 

This is the only way you could do it for now as `connectOrCreate` only supports unique fields. I would suggest creating a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with the above use case so that we can look into this 🙂

**Reply by techboy-coder** [reply]
Okay, thanks for your reply. I just created a new feature request: https://github.com/prisma/prisma/issues/7805. I'll close this discussion for now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
